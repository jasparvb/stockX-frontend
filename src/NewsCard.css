.NewsCard {
  border-radius: 4px;
  margin: 0 -30px;
  padding: 30px;
  position: relative;
  -webkit-transition: background 150ms;
  transition: background 150ms;
}

.NewsCard .box {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  min-height: 134px;
}

.NewsCard .headline-box {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.NewsCard .image-box {
  border-radius: 4px;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  height: 134px;
  width: 196px;
}

.NewsCard .image-box img {
  height: 100%;
  -o-object-fit: cover;
  object-fit: cover;
  width: 100%;
}