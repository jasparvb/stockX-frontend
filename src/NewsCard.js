import React from "react";
import './NewsCard.css';

function NewsCard({headline, image, url, source}) {
  return (
    <a href={url} target="_blank" rel="noopener noreferrer">
      <div className="NewsCard">
        <div className="box">
          <div className="headline-box">
            <p>{source}</p>
            <h4>{headline}</h4>
          </div>
          <div className="image-box">
            <img src={image} alt={headline} />
          </div>
        </div>
      </div>
    </a>
  );
}

export default NewsCard;