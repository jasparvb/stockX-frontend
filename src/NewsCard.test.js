import React from "react";
import { render } from "@testing-library/react";
import NewsCard from "./NewsCard";

it("renders without crashing", function() {
  render(<NewsCard />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<NewsCard />);
  expect(asFragment()).toMatchSnapshot();
});
