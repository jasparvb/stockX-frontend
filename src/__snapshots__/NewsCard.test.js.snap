// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="news text-dark"
    rel="noopener noreferrer"
    target="_blank"
  >
    <div
      class="NewsCard col-md-10"
    >
      <div
        class="box"
      >
        <div
          class="headline-box"
        >
          <p
            class="mb-0"
          />
          <h4 />
        </div>
        <div
          class="image-box"
        >
          <img />
        </div>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
